pytensor/link/pytorch/dispatch/random.py

```diff
@@ -0,0 +1,61 @@
+from functools import singledispatch
+
+import torch
+from numpy.random import Generator
+
+import pytensor.tensor.random.basic as ptr
+from pytensor.link.pytorch.dispatch.basic import pytorch_funcify, pytorch_typify
+
+
+@pytorch_typify.register(Generator)
+def pytorch_typify_Generator(rng, **kwargs):
+    # XXX: Check if there is a better way.
+    # Numpy uses PCG64 while Torch uses Mersenne-Twister (https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/CPUGeneratorImpl.cpp)
```

Comment:
fwiw, this depends on the device iiuc: https://open.spotify.com/episode/13oJCmQ2JWbk7t6sLRWlDz?si=1f43e67353284cc7